rust/gelato/src/fwd_req_call.rs

```diff
@@ -0,0 +1,267 @@
+use crate::err::GelatoError;
+use crate::fwd_req_op::ForwardRequestOpArgs;
+use ethers::types::Signature;
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Serialize, Serializer};
+use std::sync::Arc;
+use tracing::info;
+use tracing::instrument;
+
+const GATEWAY_URL: &str = "https://gateway.api.gelato.digital";
+
+#[derive(Debug, Clone)]
+pub struct ForwardRequestCall {
+    pub http: Arc<reqwest::Client>,
+    pub args: ForwardRequestOpArgs,
+    pub sig: Signature,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ForwardRequestCallResult {
+    pub task_id: String,
+}
+
+impl ForwardRequestCall {
+    #[instrument]
+    pub async fn run(
+        &self,
```

Comment:
I guess your motivation for that is something like: specifically we want to invalidate the call object after invocation -- if somebody wants to retry the call, they need to make a new one and then call run() on the new one?

I see the following guidance in chapter 5.3 of the rust book which makes some sense to me, just want to make sure I understand the recommenation:

""""Having a method that takes ownership of the instance by using just self as the first parameter is rare; this technique is usually used when the method transforms self into something else and you want to prevent the caller from using the original instance after the transformation."""

Reply:
Ok I've changed to self, but still wonder about the question I posed a couple weeks ago, so will leave this thread open